sdk/azidentity/README.md

```diff
@@ -0,0 +1,279 @@
+|variable name|value
+|-|-
+|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
+|`AZURE_TENANT_ID`|id of the application's Azure Active Directory tenant
+|`AZURE_CLIENT_SECRET`|one of the application's client secrets
+
+#### Service principal with certificate
+|variable name|value
+|-|-
+|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
+|`AZURE_TENANT_ID`|id of the application's Azure Active Directory tenant
+|`AZURE_CLIENT_CERTIFICATE_PATH`|path to a PEM-encoded certificate file including private key (without password protection)
+
+#### Username and password
+|variable name|value
+|-|-
+|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
+|`AZURE_USERNAME`|a username (usually an email address)
+|`AZURE_PASSWORD`|that user's password
+
+Configuration is attempted in the above order. For example, if values for a
+client secret and certificate are both present, the client secret will be used.
+
+## Troubleshooting
+
+### Error Handling
+
+Credentials return `CredentialUnavailableError` when they're unable to attempt
+authentication because they lack required data or state. For example,
+`NewEnvironmentCredential` will raise this exception when
+[its configuration](#environment-variables "its configuration") is incomplete.
+
+Credentials raise `AuthenticationFailedError` when they fail
+to authenticate. Call `Error()` on `AuthenticationFailedError` to see why authentication failed. When returned by
+`DefaultAzureCredential` or `ChainedTokenCredential`,
+the message collects error messages from each credential in the chain.
+
+For more details on handling specific Azure Active Directory errors please refer to the
+Azure Active Directory
+[error code documentation](https://docs.microsoft.com/azure/active-directory/develop/reference-aadsts-error-codes).
+
+### Logging
+
+This library uses the classification based logging implementation in azcore. To turn on logging set `AZURE_SDK_GO_LOG` to `all`. If you only want to include logs for azidentity, you must create you own logger and set the log classification as `LogCredential`.
+Credentials log basic information only, including `GetToken` success or failure and errors. These log entries do not contain authentication secrets.
+
+To obtain more detailed logging, including request/response bodies and header values, make sure to leave the logger as default or enable the `LogRequest` and/or `LogResponse` classificatons. A logger that only includes credential logs can be like the following:
+
+```go
+// Set log to output to the console
+log := azcore.Log()
+log.SetListener(func(cls LogClassification, s string) {
+		fmt.Println(s) // printing log out to the console
+  })
+  
+// Include only azidentity credential logs
+log.SetClassifications(azidentity.LogCredential)
+defer log.resetClassifications()
+
+```
+
+> CAUTION: logs from credentials contain sensitive information.
+> These logs must be protected to avoid compromising account security.
```

Comment:
It would be good to expand on this as it seems to conflict with the earlier statement "These log entries do not contain authentication secrets."

Reply:
I'm not sure about expanding specifically. Sensitive information could be email addresses which are technically not secret.

Reply:
Perhaps the other sentence needs to be augmented then?  Maybe something like "These log entries do not contain authentication secrets but might contain sensitive information."